Source/WebCore/css/values/primitives/CSSPrimitiveNumericRange.h

```diff
@@ -93,34 +94,29 @@ inline constexpr auto ClosedPercentageRangeClampUpper = Range { 0, 100, RangeOpt
 // Clamps a floating point value to within `range`.
 template<Range range, std::floating_point T> constexpr T clampToRange(T value)
 {
-    if constexpr (range.min == -Range::infinity && range.max == Range::infinity)
-        return value;
-    else if constexpr (range.max == Range::infinity)
-        return std::max<T>(value, range.min);
-    else if constexpr (range.min == -Range::infinity)
-        return std::min<T>(value, range.max);
-    else
-        return std::clamp<T>(value, range.min, range.max);
+    return clampTo<T>(
+        value,
+        std::max<T>(range.min, -std::numeric_limits<T>::max()),
+        std::min<T>(range.max,  std::numeric_limits<T>::max())
+    );
 }
 
 // Clamps a floating point value to within `range` and within additional provided range.
 template<Range range, std::floating_point T> constexpr T clampToRange(T value, T additionalMinimum, T additionalMaximum)
 {
-    return std::clamp<T>(value, std::max<T>(range.min, additionalMinimum), std::min<T>(range.max, additionalMaximum));
+    return clampTo<T>(
+        value,
+        std::max<T>(std::max<T>(range.min, -std::numeric_limits<T>::max()), additionalMinimum),
+        std::min<T>(std::min<T>(range.max,  std::numeric_limits<T>::max()), additionalMaximum)
```

Comment:
Consider something more like this?
```suggestion
        std::max<T>({ range.min, -std::numeric_limits<T>::max(), additionalMinimum }),
        std::min<T>({ range.max,  std::numeric_limits<T>::max(), additionalMaximum })
```

Reply:
I actually did the nested ones on purpose (I should add a comment explaining) as the inner ones are guaranteed constexpr but the outer one is not.